documentation/modules/ROOT/pages/03-demo.adoc

```diff
@@ -7,28 +7,879 @@
 
 == Demonstration
 
-Include here content related to potential existing demos: blogs, articles, recorded videos, walkthrough guides, tutorials.
+Let's take a look at the following example.
+We'll be looking at a Cabs ride-sharing application, that mimics real-world solutions of similar kind.
+The application is written in a popular Java framework.
 
-[#demo-video]
-=== Watch a demonstration
+=== Initial application
 
-In this video you can see xpto:
+[IMPORTANT]
+====
+The following example should be considered as suboptimal, most likely a counterexample!
+====
 
-video::3yULVMdqJ98[youtube, width=800, height=480]
+Here's a method from the Cabs application that handles the completion of a ride.
 
-Next, you can learn how to walkthrough this demo.
+[source,java]
+----
+@Transactional // <1>
```

Comment:
the java is hard to read - I'd group it into pieces?

REads easier here - in `adoc` than rendered on the preview